### ISSUE TITLE
When loading snapshots, webconf's MIDI options doesn't reflect UI state

### ISSUE THREAD

**Issue by riban-bw** [zynthian/zynthian-issue-tracking#809]
**Describe the bug**
Loading snapshot "FluidR3 GM" adds 15 chains of Fluidsynth. Attempts to add a 16th chain fail.

**To Reproduce**
Steps to reproduce the behaviour:
1. Load snapshot "FluidR3 GM"
2. Add another chain

**Expected behaviour**
Can add another chain.

**Actual behaviour**
No MIDI channels available to add chain.

**Setup:**
- HARDWARE
Raspberry Pi 4 Model B Rev 1.4
Audio: [ZynADAC](http://192.168.1.51/hw-audio)
Display: [ZynScreen 3.5 (v1)](http://192.168.1.51/hw-display)
Wiring: [MCP23017_ZynScreen](http://192.168.1.51/hw-wiring)
I2C: [MCP23017@0x20, MCP23008@0x21, ADS1115@0x48](http://192.168.1.51/hw-wiring)
- SYSTEM
Raspbian GNU/Linux 10 (buster)
Build Date: 2022-10-28
Memory: 15% (297M/1872M)
SD Card: 40% (12G/29G)
Temperature: 54.5ºC
Overclock: [Medium](http://192.168.1.51/hw-options)
- MIDI
Profile: [default.sh](http://192.168.1.51/ui-midi-options)
Tuning: [440 Hz](http://192.168.1.51/ui-midi-options)
Receive Mode: [Stage (Omni On)](http://192.168.1.51/ui-midi-options)
ZS3 Sub-SnapShots: [On](http://192.168.1.51/ui-midi-options)
MIDI to Output: [Off](http://192.168.1.51/ui-midi-options)
Master Channel: [off](http://192.168.1.51/ui-midi-options)
- SOFTWARE
zyncoder: [staging-2210 (98c3df2)](https://github.com/zynthian/zyncoder/commit/98c3df25193485902ecdd2da4a312e5a3c87843b)
zynthian-ui: [staging-2210 (62051e5)](https://github.com/zynthian/zynthian-ui/commit/62051e58d58030a7c9f55f6e3d629c8b0b12d1d7)
zynthian-sys: [staging-2210 (8c7e2ac)](https://github.com/zynthian/zynthian-sys/commit/8c7e2ac6b8a1209f73f2319f74a326d75bd03d8d)
zynthian-data: [staging-2210 (3bbc39c)](https://github.com/zynthian/zynthian-data/commit/3bbc39cb3febe968dd2be3155f547db1848dba71)
zynthian-webconf: [staging-2210 (b60b529)](https://github.com/zynthian/zynthian-webconf/commit/b60b5292d056a3966ddf105a6399ef3bd85bdba7)

**Additional context**
Is this related to master channel? Master channel is shown as disabled in webconf.

**Comment by riban-bw** [comment]
It looks like master channel is configured within the snapshot but is not updating webconf.

**Comment by riban-bw** [comment]
It looks like the master channel is stored in the snapshot which, when restored does not update webconf.

**Comment by jofemodo** [comment]
Your are right. This is an old issue and it should be solved ASAP.
When loading a snapshot, included MIDI options are not reflected in webconf, that shows the selected MIDI profile (on disk!) instead of showing the "currently-loaded MIDI options". Currently we don't have a mechanism for getting the "currently loaded MIDI options" and it should be implemented.

**Comment by riban-bw** [comment]
@jofemodo we agreed this was urgent back in 2022. Should this be sorted in the next Oram release?

**Comment by riban-bw** [comment]
I will look at this.

**Comment by riban-bw** [comment]
I don't think this is an issue since we changed to chain manager. The ability to add instruments or audio chains is limited by the quantity of audio mixer channels. The allocation of MIDI channel is limited if master channel is enabled but can assign to duplcate MIDI channels. FluidR3 GM" does load with all 16 chains, even if master channel is enabled but chain 16 has no MIDI channels selected. Mixer erroneously shows MIDI channel 16 is selected but I am not going to fix this. It is minor and few will even notice. A future change to mixer may change this behaviour anyway.
